### PR TITLE
Rework soul credits and migrations

### PR DESCRIPTION
**Pull request by TheRedPanda17**

### CODE REVIEW

**Review by rj-lee — APPROVED** [review]

**Comment by TheRedPanda17** [comment]
@rj-lee Were you able to test it locally?

File: myning/chapters/enter_graveyard.py [hunk 1]
```diff
@@ -82,8 +81,9 @@ def revive(player: Player, ally: Character):
         FileManager.multi_save(player, ally)
 
 
-def soul_cost(ghost_count: int):
-    return int(fibonacci(ghost_count + 1))
+@cache
```
Comment (jeremyd85) [inline myning/chapters/enter_graveyard.py:84]: Why cache?

Comment (jeremyd85) [reply myning/chapters/enter_graveyard.py:84]: I guess it doesn't hurt actually. It just seems like a cheap calculation to cache. I approve and I will now migrate to GitHub :)

**Comment by rj-lee** [comment]
> @rj-lee Were you able to test it locally?

```
You've spent 1 soul credits.

New cost is 3 soul credits.

You've been returned -2 soul credits.

Have fun myning!

MIGRATION 2 RAN (Enter to continue)
```

I've resurrected two people. RIP -2 soul credits

**Comment by jeremyd85** [comment]
I got +9 soul credits. It was quite good for me :)

File: myning/chapters/enter_graveyard.py (hunk 1 above)
Comment (TheRedPanda17) [reply myning/chapters/enter_graveyard.py:84]: You're right. I cached it when it used to be recursive. I'll remove it.

File: myning/chapters/enter_graveyard.py [hunk 2]
```diff
@@ -1,11 +1,10 @@
-from math import sqrt
+from functools import cache
```
Comment (rj-lee) [inline myning/chapters/enter_graveyard.py:1]: could probably remove this then since it's unused

File: myning/migrations/soul_credits.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    returned = spent_credits - new_cost
+    if returned > 0:
+        print(f"\nYou've been returned {get_soul_string(returned)}.")
+        player.soul_credits += returned
+        FileManager.save(player)
+    elif returned == 0:
+        print(f"\nYou broke even!")
+    else:
+        print("You actually came out ahead before, so we'll give you a pass.")
```
Comment (rj-lee) [inline myning/migrations/soul_credits.py:31]: not sure if you wanted this to have a `\n` in front. doesn't really matter to me

**Review by rj-lee — APPROVED** [review]
lint/format github actions soon?

File: myning/chapters/enter_graveyard.py (hunk 2 above)
Comment (TheRedPanda17) [reply myning/chapters/enter_graveyard.py:1]: I'll get it in the next MR, coming soon.

**Pull request merged**